### PR TITLE
Bumped image versions to v4.0.0

### PR DESCRIPTION
**Pull request by kolbe**
### What problem does this PR solve? 
Bumped image versions from 3.0.13 to 4.0.0.

### Check List 

Tests 

 - No code

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
TiDB Operator examples updated to deploy TiDB Cluster 4.0.0 images.
```

### CODE REVIEW

File: charts/tidb-cluster/values.yaml [hunk 1]
```diff
@@ -108,7 +108,7 @@ pd:
     # annotations: {} "<the default name will be used>"
     # portName: "client"
   replicas: 3
-  image: pingcap/pd:v3.0.13
+  image: pingcap/pd:v4.0.0
```
Comment (Yisaer) [inline charts/tidb-cluster/values.yaml:111]: The helm chart for tidbcluster is no longer maintained after operator 1.1 released. I'm not sure whether we should update it.

File: charts/tidb-drainer/values.yaml [hunk 2]
```diff
@@ -12,7 +12,7 @@ timezone: UTC
 
 # clusterName is the TiDB cluster name that should backup from or restore to.
 clusterName: demo
-clusterVersion: v3.0.13
+clusterVersion: v4.0.0
```
Comment (Yisaer) [inline charts/tidb-drainer/values.yaml:15]: ditto.

File: charts/tidb-lightning/values.yaml [hunk 3]
```diff
@@ -5,7 +5,7 @@
 # timezone is the default system timzone
 timezone: UTC
 
-image: pingcap/tidb-lightning:v3.0.13
+image: pingcap/tidb-lightning:v4.0.0
```
Comment (Yisaer) [inline charts/tidb-lightning/values.yaml:8]: ditto.

File: charts/tikv-importer/values.yaml [hunk 4]
```diff
@@ -8,7 +8,7 @@ timezone: UTC
 # clusterName is the TiDB cluster name, if not specified, the chart release name will be used
 clusterName: demo
 
-image: pingcap/tidb-lightning:v3.0.13
+image: pingcap/tidb-lightning:v4.0.0
```
Comment (Yisaer) [inline charts/tikv-importer/values.yaml:11]: ditto.

**Comment by cofyc** [comment]
I think it's ok to bump the tidb version in these charts.

**Comment by cofyc** [comment]
btw, `tikv-importer` is introduced in v1.1 to deploy tidb-lightning. We may still use helm charts to do simple tasks.

**Comment by cofyc** [comment]
The `TiDBCluster Backup and restore TiDB Cluster` test fails again.

https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/operator_ghpr_e2e_test_kind/detail/operator_ghpr_e2e_test_kind/6352/tests

maybe we have some issues must be fixed

**Comment by cofyc** [comment]
cc @onlymellb @shuijing198799 can you look at this PR?

**Pull request merged**